rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -3553,12 +3568,24 @@ static rmw_ret_t rmw_take_ser_int(
         serialized_message->buffer_length = size;
         ddsi_serdata_unref(d);
         *taken = true;
+        TRACETOOLS_TRACEPOINT(
+          rmw_take,
+          static_cast<const void *>(subscription),
+          static_cast<const void *>(serialized_message),
+          (message_info ? message_info->source_timestamp : 0LL),
+          *taken);
         return RMW_RET_OK;
       }
     }
     ddsi_serdata_unref(d);
   }
   *taken = false;
+  TRACETOOLS_TRACEPOINT(
+    rmw_take,
+    static_cast<const void *>(subscription),
+    static_cast<const void *>(serialized_message),
+    (message_info ? message_info->source_timestamp : 0LL),
```

Comment:
If nothing was taken, `message_info->source_timestamp` is meaningless.

Wouldn't:
```suggestion
    0LL,
```
make more sense?

That is, of course, under the assumption that you really want the tracepoint to fire when successfully taking nothing, rather than firing only when actually taking something.

Reply:
Thank you for your review, I required.